Objects/bytesobject.c

```diff
@@ -463,8 +464,11 @@ formatfloat(PyObject *v, int flags, int prec, int type,
     if (prec < 0)
         prec = 6;
 
-    p = PyOS_double_to_string(x, type, prec,
-                              (flags & F_ALT) ? Py_DTSF_ALT : 0, NULL);
+    if (flags & F_ALT)
+        dtoa_flags |= Py_DTSF_ALT;
+    if (flags & F_NO_NEG_0)
+        dtoa_flags |= Py_DTSF_NO_NEG_0;
```

Comment:
PEP 7 nitpicking (also applies to changes `unicodeobject.c`, though it's not always obvious what counts as "new code" for the purposes of PEP 7's "All new C code requires braces"):
```suggestion
    if (flags & F_ALT) {
        dtoa_flags |= Py_DTSF_ALT;
    }
    if (flags & F_NO_NEG_0) {
        dtoa_flags |= Py_DTSF_NO_NEG_0;
    }
```